operational/azure/azuread_group_sync/azuread_group_sync.pt

```diff
@@ -0,0 +1,754 @@
+EOS
+end
+
+script "js_filtered_grs_groups", type: "javascript" do
+  parameters "ds_filtered_aad_groups", "ds_grs_groups"
+  result "results"
+  code <<-EOS
+    var results = {};
+    var groupsToCreate = [];
+    var groupsToUpdate = [];
+    
+    // Filter groups to be updated/created
+    var groupsNotInGRS = _.difference(_.pluck(ds_filtered_aad_groups, "displayName"), _.pluck(ds_grs_groups, "name"))
+    var groupsOnlyInGRS = _.difference(_.pluck(ds_grs_groups, "name"), _.pluck(ds_filtered_aad_groups, "displayName"))
+    var groupsInGRS = _.difference(_.pluck(ds_grs_groups, "name"), groupsOnlyInGRS)
+
+    _.each(groupsNotInGRS, function(group) {
+      var currentGroupObject = _.where(ds_filtered_aad_groups, {displayName: group});
+      var aadGroupObject = _.where(ds_filtered_aad_groups, {displayName: group});
+
+      groupsToCreate.push({
+        name: currentGroupObject[0]["displayName"],
+        id: "N/A",
+        href: "N/A",
+        action: "Create",
+        objectId: aadGroupObject[0]["objectId"]
+      })
+    })
+
+    _.each(groupsInGRS, function(group) {
+      var currentGroupObject = _.where(ds_grs_groups, {name: group});
+      var aadGroupObject = _.where(ds_filtered_aad_groups, {displayName: group});
+      var aadUserCount = _.size(aadGroupObject[0]["members"])
+      cmpUserCount = _.size(currentGroupObject[0]["users"])
+
+      groupsToUpdate.push({
+        name: currentGroupObject[0]["name"],
+        id: currentGroupObject[0]["id"],
+        href: currentGroupObject[0]["href"],
+        action: "Update",
+        aadUserCount: aadUserCount,
+        cmpUserCount: cmpUserCount
+        objectId: aadGroupObject[0]["objectId"]
+      })
+    })
+
+    results = {
+      GroupsToCreate: groupsToCreate,
+      GroupsToUpdate: groupsToUpdate
+    }
+EOS
+end
+
+script "js_combined_data", type: "javascript" do
+  parameters "ds_filtered_aad_groups", "ds_filtered_aad_users", "ds_filtered_grs_users", "ds_filtered_grs_groups", "ds_grs_groups", "ds_grs_users"
+  result "results"
+  code <<-EOS
+    var results = {};
+    results= {
+      aadGroups: ds_filtered_aad_groups,
+      aadUsers: ds_filtered_aad_users,
+      grsGroups: ds_grs_groups,
+      grsUsers: ds_grs_users,
+      groupsToCreate: ds_filtered_grs_groups["GroupsToCreate"],
+      groupsToUpdate: ds_filtered_grs_groups["GroupsToUpdate"],
+      usersToCreate: ds_filtered_grs_users["UsersToCreate"],
+      usersToPurge: ds_filtered_grs_users["UsersToPurge"],
+    }
+EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_azuread_group_sync" do
+  validate $ds_combined_data do
+    summary_template "{{ len data.groupsToUpdate }} AzureAD Groups to Synchronize"
+    detail_template <<-EOS
+## Groups that Will Be Synchronized from AzureAD
+| Name  | Href  | AzureAD Object Id | AzureAD User Count | CMP User Count |
```

Comment:
sort by name?

Reply:
I believe the AzureAD API returns results sorted. Are you asking that I forcibly sort it in the datasource just to be sure?

Reply:
Added a sort to the groups to update and create.

Reply:
I didn't review the data, as long as the data is sorted in a logical way that's all we needed.  most people have been forcing with _sortBy()